### PR TITLE
http_outlet: retry on more errors

### PR DESCRIPTION
**Pull request by danp**
Retry on all errors returned by post, even if they are not of type *url.Error.

### CODE REVIEW

File: http_outlet_test.go [hunk 1]
```diff
@@ -79,7 +82,7 @@ func TestOutletEOFRetryMax(t *testing.T) {
 	config.SkipVerify = true
 	logCapture := new(bytes.Buffer)
 	s := NewShuttle(config)
-	s.ErrLogger = log.New(logCapture, "", 0)
+	s.ErrLogger = log.New(io.MultiWriter(logCapture, os.Stdout), "", 0)
```
Comment (freeformz) [inline http_outlet_test.go]: Was this there only for reviewing test output?

File: http_outlet_test.go [hunk 2]
```diff
@@ -89,23 +92,98 @@ func TestOutletEOFRetryMax(t *testing.T) {
...
+	}))
+	defer ts.Close()
+
+	config := newTestConfig()
+	config.LogsURL = ts.URL
+	config.SkipVerify = true
+
+	logCapture := new(bytes.Buffer)
+	s := NewShuttle(config)
+	s.ErrLogger = log.New(io.MultiWriter(logCapture, os.Stdout), "", 0)
```
Comment (freeformz) [inline http_outlet_test.go]: I'd much rather t.Info the buffer.Bytes()

File: http_outlet_test.go [hunk 3]
```diff
@@ -121,9 +199,9 @@ func TestTimeout(t *testing.T) {
 	config.SkipVerify = true
 	config.Timeout = 100 * time.Millisecond
 
+	logCapture := new(bytes.Buffer)
```
Comment (freeformz) [inline http_outlet_test.go]: It's probably not consistent in the existing code, but I prefer `var logCapture bytes.Buffer; &logCapture`

**Comment by freeformz** [comment]
Other then some comments, looks good to me. 👍

File: http_outlet_test.go (hunk 2 above)
Comment (danp) [reply http_outlet_test.go]: Went ahead and dropped the extra output altogether, can address some other time.

File: http_outlet_test.go (hunk 3 above)
Comment (danp) [reply http_outlet_test.go]: Done, and tried to get some more consistency around that.

**Comment by amerine** [comment]
Excellent patch! Thanks, @danp

**Pull request merged**